tests/codeceptjs/steps_file.js

```diff
@@ -7,6 +7,12 @@ module.exports = function() {
 
     // Define custom steps here, use 'this' to access default methods of I.
     // It is recommended to place a general 'login' function here.
-
+    pressKeys: function (string) {
```

Comment:
Shouldn't we place this in `config/codeceptjs_helpers.js` so all the custom functions is located in one place?

Reply:
Yes. @germanbisurgi Can you please move it.

Reply:
about that, why do we have an extra codeceptjs_helpers?

Reply:
https://github.com/json-editor/json-editor/blob/cf84e2cd3e332d7f4683fbbc39ad55451a41b777/config/codeceptjs_helpers.js

Reply:
@pmk65 We should move that stuff to `tests`, btw.

Reply:
> about that, why do we have an extra codeceptjs_helpers file?

Where would you have put it?

Reply:
AFIK. There's a difference between `Helpers` and `Steps` 
In Steps `this` is the current helper, where you in Helpers have do define it yourself.
Helpers also work regardless of the steps option is enabled or not.

Reply:
in steps_file.js in the test/codeceptjs folder. This file is automatic generated when setting codecept js with the only purpose to extend the actor object of codeceptjs. 

Also why do we need puppeteer? we had already tests for chrome and firefox with selenium

Reply:
pupeteer is AFAIK for local testing without Docker (@pmk65 right?)

Reply:
>
> Also why do we need puppeteer? we had already tests for chrome and firefox with selenium

Because not everyone can run Docker. 😁

Reply:
> 
> 
> pupeteer is AFAIK for local testing without Docker (@pmk65 right?)

Yes. If you got a lame Windows 10 Home edition (like me 😝), Docker won't work, but Puppeteer does.

Reply:
ok but

- with Puppeteer, Installation and usage will change. https://github.com/GoogleChrome/puppeteer/tree/master/experimental/puppeteer-firefox
- There are npm packages for selenium (firefox and chrome)

I will move the code to the helper file

Reply:
You just can’t use the Hyper-V version of it but you can use the VirtualBox version. https://thewebspark.com/2019/04/04/how-to-install-docker-on-windows-10-home-edition-solved/

Reply:
> You just can’t use the Hyper-V version of it but you can use the VirtualBox version. https://thewebspark.com/2019/04/04/how-to-install-docker-on-windows-10-home-edition-solved/

@pmk65 You'll come to the point where answering these questions is more painful than acutally installing Docker :joy::joy::joy:
@germanbisurgi I tried it very hard to bring Peter to installing Docker somehow.